lib/hooks/useResources.ts

```
@@ -0,0 +1,200 @@
+
+  // Helper function to filter resources based on search query
+  const filterResources = useCallback(
+    (resources: Resource[], query: string) => {
+      if (!query) {
+        setFilteredResources(resources);
+      } else {
+        const filtered = resources.filter(
+          (resource) =>
+            resource.title.toLowerCase().includes(query.toLowerCase()) ||
+            (resource.moduleName &&
+              resource.moduleName.toLowerCase().includes(query.toLowerCase()))
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)